src/FluxCalculator.jl

```diff
@@ -439,4 +382,95 @@ function water_albedo_from_atmosphere!(atmos_sim::Interfacer.AtmosModelSimulatio
     error("this function is required to be dispatched on" * Interfacer.name(atmos_sim) * ", but no method defined")
 end
 
+"""
+    compute_surface_fluxes!(csf, sim, atmos_sim, boundary_space, thermo_params, surface_scheme)
+
+This function computes surface fluxes between the input component model
+simulation and the atmosphere.
+This is intended to be used with the partitioned fluxes option, and
+should be extended for any model that requires non-standard flux calculations.
+
+Update the input coupler surface fields `csf` in-place with the computed fluxes
+for this model. These are then summed using area-weighting across all surface
+models to get the total fluxes.
+
+Since the fluxes are computed between the input model and the atmosphere, this
+function does nothing if called on an atmosphere model simulation.
+
+# Arguments
+- `csf`: [CC.Fields.Field] containing a NamedTuple of turbulent flux fields: `F_turb_ρτxz`, `F_turb_ρτyz`, `F_turb_energy`, `F_turb_moisture`.
+- `sim`: [Interfacer.ComponentModelSimulation] the surface simulation to compute fluxes for.
+- `atmos_sim`: [Interfacer.AtmosModelSimulation] the atmosphere simulation to compute fluxes with.
+- `boundary_space`: [CC.Spaces.AbstractSpace] the space of the coupler surface.
+- `thermo_params`: [TD.Parameters.ThermodynamicsParameters] the thermodynamic parameters.
+- `surface_scheme`: [AbstractSurfaceFluxScheme] the surface flux scheme.
+"""
+function compute_surface_fluxes!(
+    csf,
+    sim::Interfacer.AtmosModelSimulation,
```

Comment:
Are `sim` and `atmos_sim` really the same type? or should the sim be the surface simulation

Reply:
oh maybe this just simplifies the loop 👍

Reply:
Yeah exactly